### DISCUSSION TITLE
Is Community.VisualStudio.Toolkit.DependencyInjection.Microsoft.17 size increase really that large?

### DISCUSSION THREAD

**Discussion by aglasencnik** [VsixCommunity/Community.VisualStudio.Toolkit#466] [Q&A]
Hi, I wanted to include dependency injection in my visual studio extension, but when I installed the: Community.VisualStudio.Toolkit.DependencyInjection.Microsoft.17 nuget package and built the project I saw a few things that concern me... First there appeared way way way more dll files that were there before... before (I was testing without any custom files just empty extension) it was just 400KB with the Community toolkit package, but now it is over 26MB... is this normal? Would this impact the extensions performance?
Also in the build window I got the following messages:
```
1>------ Rebuild All started: Project: TestBuildExtension, Configuration: Debug Any CPU ------
Restored C:\Users\Amadej\Desktop\TestBuildExtension\TestBuildExtension\TestBuildExtension.csproj (in 135 ms).
1>  TestBuildExtension -> C:\Users\Amadej\Desktop\TestBuildExtension\TestBuildExtension\bin\Debug\TestBuildExtension.dll
1>  TestBuildExtension -> C:\Users\Amadej\Desktop\TestBuildExtension\TestBuildExtension\bin\Debug\TestBuildExtension.vsix
1>C:\Users\Amadej\.nuget\packages\microsoft.vssdk.compatibilityanalyzer\17.7.2196\build\Microsoft.VsSDK.CompatibilityAnalyzer.targets(36,5): warning VSIXCompatibility1001: The extension is incompatible with the targeted version of Visual Studio. More info at https://aka.ms/ExtensionSdkErrors. 
1>C:\Users\Amadej\.nuget\packages\microsoft.vssdk.compatibilityanalyzer\17.7.2196\build\Microsoft.VsSDK.CompatibilityAnalyzer.targets(36,5): warning VSIXCompatibility1001: Following references are incompatible.
1>C:\Users\Amadej\.nuget\packages\microsoft.vssdk.compatibilityanalyzer\17.7.2196\build\Microsoft.VsSDK.CompatibilityAnalyzer.targets(36,5): warning VSIXCompatibility1001: Microsoft.VisualStudio.Web.BrowserLink.12.0.dll -> envdte
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========
========== Rebuild started at 14:39 and took 04,524 seconds ==========
```
I got some compatibility warnings... Are they fake? Or are they actual problems?

**Comment by ErikEJ** [comment]
Have you seen https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/issues/432 - there could be multiple issues at play here

**Reply by aglasencnik** [reply]
Hi, thank you for your response!
I can see that you had a similar issue...
But the problem in my case is this only appeared when installing the DI package, before it wasn't happening...

**Reply by ErikEJ** [reply]
Open the vsix file with a zip tool and have a look?

**Reply by aglasencnik** [reply]
I mean there are like 134 files and a lot of folders which also have dll's inside them

**Reply by aglasencnik** [reply]
Should I just downgrade the nuget package version? So to the .4. ?

**Reply by aglasencnik** [reply]
Or what did you do to solve that?

**Reply by aglasencnik** [reply]
Ok so update: I downgraded the build tools to .4. And there isn't a warning anymore... Still there are a LOT of file in the debug folder, but in the .vsix it is 700KB which should be perfect....

**Reply by ErikEJ** [reply]
Yeah, I am surprised the team still publishes these broken build tools versions that are producing bad .vsix files @AArnott @madskristensen ??

**Reply by aglasencnik** [reply]
Ok so I downgraded both build tools and toolkit packages so I don't get the error anymore, but still for some reason the InitializeServices method doesn't run (I have a breakpoint) set.... Do you maybe know what I could check?...
```csharp
[ProvideAutoLoad(UIContextGuids.NoSolution, PackageAutoLoadFlags.BackgroundLoad)]
[ProvideAutoLoad(UIContextGuids.SolutionExists, PackageAutoLoadFlags.BackgroundLoad)]
public sealed class EmmetVSPackage : MicrosoftDIToolkitPackage<EmmetVSPackage>
{
    /// <summary>
    /// Initializes services.
    /// </summary>
    /// <param name="services">IServiceCollection</param>
    protected override void InitializeServices(IServiceCollection services)
    {
        try
        {
            base.InitializeServices(services);

            services.AddSingleton<IAbbreviationService, AbbreviationService>();
            services.AddSingleton<IActionUtilsService, ActionUtilsService>();
            services.AddSingleton<ICssMatcherService, CssMatcherService>();
            services.AddSingleton<IHtmlMatcherService, HtmlMatcherService>();
            services.AddSingleton<IMathExpressionService, MathExpressionService>();
            services.AddSingleton<IScannerService, ScannerService>();

            services.RegisterCommands(ServiceLifetime.Singleton);
        }
        catch (Exception ex)
        {
            ex.Log();
        }
    }

    /// <summary>
    /// Initializes the package.
    /// </summary>
    /// <param name="cancellationToken">CancellationToken object</param>
    /// <param name="progress">IProgress of ServiceProgressData object</param>
    /// <returns>A task that represents the asynchronous operation.</returns>
    protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
    {
        try
        {
            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
        }
        catch (Exception ex)
        {
            await ex.LogAsync();
        }
    }
}
```
I also have the autoload set so it loads when Visual Studio starts....

**Reply by aglasencnik** [reply]
Ok I found out that I need to add `await base.InitializeAsync();` now it runs...

**Comment by RobertvanderHulst** [comment]
@aglasencnik Calling the parent method from an overridden method is good practice!
